modules/targets/generic-linux.nix

```diff
@@ -4,49 +4,42 @@ with lib;
 
 let
 
+  cfg = config.targets.genericLinux;
+
   profileDirectory = config.home.profileDirectory;
 
 in {
+  imports = [
+    (mkRenamedOptionModule [ "targets" "genericLinux" "extraXdgDataDirs" ] [
+      "xdg"
+      "systemDirs"
+      "data"
+    ])
+  ];
+
   options.targets.genericLinux = {
     enable = mkEnableOption "" // {
       description = ''
         Whether to enable settings that make Home Manager work better on
         GNU/Linux distributions other than NixOS.
       '';
     };
-
-    extraXdgDataDirs = mkOption {
-      type = types.listOf types.str;
-      default = [ ];
-      example = [ "/usr/share" "/usr/local/share" ];
-      description = ''
-        List of directory names to add to <envar>XDG_DATA_DIRS</envar>.
-      '';
-    };
   };
 
-  config = mkIf config.targets.genericLinux.enable {
-    home.sessionVariables = let
-      profiles =
-        [ "\${NIX_STATE_DIR:-/nix/var/nix}/profiles/default" profileDirectory ];
-      dataDirs = concatStringsSep ":"
-        (map (profile: "${profile}/share") profiles
-          ++ config.targets.genericLinux.extraXdgDataDirs);
+  config = mkIf cfg.enable {
+    xdg.systemDirs.data = [
+      # Nix profiles
+      "\${NIX_STATE_DIR:-/nix/var/nix}/profiles/default/share"
+      "${profileDirectory}/share"
 
-      # https://github.com/archlinux/svntogit-packages/blob/packages/ncurses/trunk/PKGBUILD
-      # https://salsa.debian.org/debian/ncurses/-/blob/master/debian/rules
-      # https://src.fedoraproject.org/rpms/ncurses/blob/main/f/ncurses.spec
-      # https://gitweb.gentoo.org/repo/gentoo.git/tree/sys-libs/ncurses/ncurses-6.2-r1.ebuild
-      distroTerminfoDirs = concatStringsSep ":" [
-        "/etc/terminfo" # debian, fedora, gentoo
-        "/lib/terminfo" # debian
-        "/usr/share/terminfo" # package default, all distros
-      ];
-    in {
-      XDG_DATA_DIRS = "${dataDirs}\${XDG_DATA_DIRS:+:}$XDG_DATA_DIRS";
-      TERMINFO_DIRS =
-        "${profileDirectory}/share/terminfo:$TERMINFO_DIRS\${TERMINFO_DIRS:+:}${distroTerminfoDirs}";
-    };
+      # FHS share dirs
+      "/usr/share"
+      "/usr/local/share"
```

Comment:
Shouldn't `/usr/local/share` have precedence over `/usr/share`?

Reply:
Should the `xdg.systemDirs` options be `types.dag` then, if we need to control precedence? Or is there a simpler type for ordered lists?

Reply:
Using a list is fine I think.
Worst case a user can `mkForce` and override everything.